applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,302 @@
+
+		led->streaming = true;
+
+		send_data_from_queue(led);
+	}
+}
+
+static void handle_config_event(const struct config_event *event)
+{
+	if (!event->store_needed) {
+		/* Accept only events coming from transport. */
+		return;
+	}
+
+	if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+		/* Only LED STREAM events. */
+		return;
+	}
+
+	switch (TYPE_FIELD_GET(event->id)) {
+	case STREAM_DATA:
+		handle_incoming_step(event);
+		break;
+
+	default:
+		LOG_WRN("Unknown LED STREAM config event");
+		break;
+	}
+}
+
+static void handle_config_fetch_request_event(
+	const struct config_fetch_request_event *event)
+{
+	LOG_INF("Handle config fetch request event");
+
+	if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+		/* Only LED STREAM events. */
+		return;
+	}
+
+	size_t led_id = MOD_FIELD_GET(event->id);
+
+	if (led_id < 0 && led_id >= CONFIG_DESKTOP_LED_COUNT) {
+		LOG_WRN("Wrong LED id");
+		return;
+	}
+
+	struct led *led = &leds[led_id];
+
+	u8_t free_places = count_free_places(led);
+	u8_t is_ready = initialized;
+
+	LOG_INF("Free places left: %" PRIu8 " for led: %zu", free_places, led_id);
+
+	static const size_t size = sizeof(free_places) + sizeof(is_ready);
+
+	BUILD_ASSERT_MSG(size == FETCH_CONFIG_SIZE, "");
+
+	struct config_fetch_event *fetch_event = new_config_fetch_event(size);
+
+	fetch_event->id = event->id;
+	fetch_event->recipient = event->recipient;
+	fetch_event->channel_id = event->channel_id;
+
+	size_t pos = 0;
+
+	fetch_event->dyndata.data[pos] = free_places;
+	pos += sizeof(free_places);
+	fetch_event->dyndata.data[pos] = is_ready;
+
+	EVENT_SUBMIT(fetch_event);
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_config_event(eh)) {
+		handle_config_event(cast_config_event(eh));
+		return false;
+	}
+
+	if (is_config_fetch_request_event(eh)) {
+		handle_config_fetch_request_event(cast_config_fetch_request_event(eh));
+		return false;
+	}
+
+	if (is_led_event(eh)) {
+		const struct led_event *event = cast_led_event(eh);
+
+		struct led *led = &leds[event->led_id];
+
+		if (event->led_effect != &led->led_stream_effect) {
+			__ASSERT_NO_MSG(event->led_effect);
+
+			led->state_effect = event->led_effect;
+
+			if (led->streaming) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	if (is_led_ready_event(eh)) {
+		const struct led_ready_event *event = cast_led_ready_event(eh);
+
+		struct led *led = &leds[event->led_id];
+
+		if (event->led_effect == &led->led_stream_effect) {
+			send_data_from_queue(led);
+		}
+
+		return true;
+	}
+
+	if (is_module_state_event(eh)) {
+		const struct module_state_event *event =
+			cast_module_state_event(eh);
+
+		if (check_state(event, MODULE_ID(leds), MODULE_STATE_READY)) {
+			if (!initialized) {
+				initialized = true;
+				module_set_state(MODULE_STATE_READY);
+			}
+		}
+
+		if (check_state(event, MODULE_ID(leds), MODULE_STATE_OFF) ||
+		    check_state(event, MODULE_ID(leds), MODULE_STATE_STANDBY)) {
+			initialized = false;
```

Comment:
we also need to update out own state
`module_set_state(MODULE_STATE_OFF);`

Reply:
done